search/engine.go

```diff
@@ -418,30 +515,49 @@ func (e *ESEngine) DoSearch(ctx context.Context, query Query, sortBy string, fro
 		return nil, errors.Wrap(err, "ES error.")
 	}
 
-	if len(mr.Responses) != 2*len(query.LanguageOrder) {
+	if len(mr.Responses) != indexesCount*len(query.LanguageOrder) {
 		return nil, errors.New(fmt.Sprintf("Unexpected number of results %d, expected %d",
-			len(mr.Responses), 2*len(query.LanguageOrder)))
+			len(mr.Responses), indexesCount*len(query.LanguageOrder)))
 	}
 
 	// Interleave content units and collection results by language.
 	// Then go over responses and choose first not empty retults list.
 	for i := 0; i < len(query.LanguageOrder); i++ {
-		cuR := mr.Responses[i]
-		cR := mr.Responses[i+len(query.LanguageOrder)]
-		if cuR.Error != nil {
-			log.Warnf("%+v", cuR.Error)
-			return nil, errors.New("Failed multi get.")
+
+		var cuR *elastic.SearchResult
+		var cR *elastic.SearchResult
+		var sR *elastic.SearchResult
+		var joinedR *elastic.SearchResult
+
+		if indexesCount >= 2 {
```

Comment:
Use only the len(requests) here and make this code generic.
Don't split to:
```
var cuR *elastic.SearchResult
var cR *elastic.SearchResult
var sR *elastic.SearchResult
```

instead use
`var results []*elastic.SearchResult`

Reply:
Issue created AS-36